tensorboard/plugins/hparams/list_session_groups.py

```diff
@@ -263,163 +234,385 @@ def _create_response(self, session_groups):
-    assert isinstance(interval, api_pb2.Interval)
-    self._interval = interval
+def _create_interval_filter(interval):
+  """Returns a function that checkes whether a number belongs to an interval.
 
-  def _value_passes(self, value):
+  Args:
+    interval: A tensorboard.hparams.Interval protobuf describing the interval.
+  Returns:
+    A function taking a number (a float or an object of a type in
+    six.integer_types) that returns True if the number belongs to (the closed)
+    'interval'.
+  """
+  def filter_fn(value):
     if (not isinstance(value, six.integer_types) and
         not isinstance(value, float)):
       raise error.HParamsError(
           'Cannot use an interval filter for a value of type: %s, Value: %s' %
           (type(value), value))
-    return (self._interval.min_value <= value and
-            value <= self._interval.max_value)
+    return interval.min_value <= value and value <= interval.max_value
+
+  return filter_fn
+
+
+def _create_discrete_set_filter(discrete_set):
+  """Returns a function that checks whether a value belongs to a set.
+
+  Args:
+    discrete_set: A list of objects representing the set.
+  Returns:
+    A function taking an object and returns True if its in the set. Membership
+    is tested using the Python 'in' operator (thus, equality of distinct
+    objects is computed using the '==' operator).
+  """
+  def filter_fn(value):
+    return value in discrete_set
 
+  return filter_fn
 
-class _SessionGroupDiscreteSetFilter(_SessionGroupFilter):
-  def __init__(self, discrete_set, extractor, include_missing_values):
-    (super(_SessionGroupDiscreteSetFilter, self)
-     .__init__(extractor, include_missing_values))
-    self._discrete_set = set(discrete_set)
 
-  def _value_passes(self, value):
-    return value in self._discrete_set
+def _value_to_python(value):
+  """Converts a google.protobuf.Value to a native Python object."""
+
+  assert isinstance(value, struct_pb2.Value)
+  field = value.WhichOneof('kind')
+  if field == 'number_value':
+    return value.number_value
+  elif field == 'string_value':
+    return value.string_value
+  elif field == 'bool_value':
+    return value.bool_value
+  else:
+    raise ValueError('Unknown struct_pb2.Value oneof field set: %s' % field)
+
+
+# As protobuffers are mutable we can't use MetricName directly as a dict's key.
+# Instead, we duplicate MetricName as a collections.namedtuple. Another
+# alternative would be to wrap the MetricName protocol buffer class in an
+# immutable class that defines equality and __hash__ methods based on the text
+# representation of the protocol buffer. This is more complex, but won't
+# require modification if we ever add fields to MetricName.
+_MetricIdentifier = collections.namedtuple('_MetricIdentifier', 'group tag')
+
+
+# We use slots here to catch typos in attributes earlier. Note that this makes
+# this class incompatible with 'pickle'.
+@attr.s(slots=True)
```

Comment:
Is there a way to do this without the new dependency? Team seem to be against this at least for this use case.